salsa/tests/chacha20.rs

```diff
@@ -0,0 +1,290 @@
+extern crate stream_cipher;
+extern crate block_cipher_trait;
+extern crate salsa;
+
+use block_cipher_trait::generic_array::GenericArray;
+use salsa::ChaCha20;
+use stream_cipher::NewStreamCipher;
+use stream_cipher::StreamCipher;
+use stream_cipher::SyncStreamCipherSeek;
+
+#[cfg(test)]
+const KEY_BYTES: usize = 32;
+
+#[cfg(test)]
+const IV_BYTES: usize = 8;
+
+#[cfg(test)]
+const IETF_KEY0: [u8; KEY_BYTES] = [0; KEY_BYTES];
+
+#[cfg(test)]
+const IETF_KEY1: [u8; KEY_BYTES] =
+    [ 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x01 ];
```

Comment:
Not sure why @newpavlov isn't responding, but I think he'd probably suggest using `.blb` files for the test vectors ala:

https://github.com/RustCrypto/stream-ciphers/tree/master/aes-ctr/tests/data

and the test macros here:

https://github.com/RustCrypto/stream-ciphers/tree/master/aes-ctr/tests/data

@emc2 if you do that, unless I hear from @newpavlov not to I don't see any reason why this can't be merged (and I have merge access)